lib/zipkin-tracer/sidekiq/middleware.rb

```diff
@@ -0,0 +1,52 @@
+module ZipkinTracer
+  module Sidekiq
+    class Middleware
+      attr_reader :config, :tracer, :traceable_workers
+
+      def initialize(config)
+        @config = Config.new(nil, config).freeze
+        @tracer = TracerFactory.new.tracer(@config)
+        @traceable_workers = config.fetch(:traceable_workers, [])
+      end
+
+      def call(worker, job, queue, &block)
+        return block.call unless traceable_worker?(worker)
+
+        trace(worker, job, queue, &block)
+      end
+
+      private
+
+      def traceable_worker?(worker)
+        traceable_workers.include?(:all) || traceable_workers.include?(worker_name(worker))
+      end
+
+      def trace(worker, job, queue, &block)
+        trace_id = TraceGenerator.new.next_trace_id
+        result = nil
+        span_name = worker_name(worker)
+
+        TraceContainer.with_trace_id(trace_id) do
+          if sample?
+            tracer.with_new_span(trace_id, span_name) do
+              result = block.call
+            end
+          else
+            result = block.call
+          end
+        end
+
+        tracer.flush!
+        result
+      end
+
+      def sample?
+        rand < config.sample_rate
```

Comment:
isn't there a way to use the usual `trace_id.sampled?` method instead of re-implementing this logic again?
see https://github.com/openzipkin/zipkin-ruby/blob/master/lib/zipkin-tracer/rack/zipkin-tracer.rb#L28-L30

If not, I guess a comment explaining why would be helpful.